instrumentation/aws-sdk/aws-sdk-1.11/library/src/main/java/io/opentelemetry/instrumentation/awssdk/v1_11/DynamoDbAttributesExtractor.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.awssdk.v1_11;
+
+import com.amazonaws.Request;
+import com.amazonaws.Response;
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.instrumentation.api.instrumenter.AttributesExtractor;
+import io.opentelemetry.instrumentation.api.internal.AttributesExtractorUtil;
+import java.util.Collections;
+import java.util.List;
+import javax.annotation.Nullable;
+
+public class DynamoDbAttributesExtractor implements AttributesExtractor<Request<?>, Response<?>> {
+
+  private static final AttributeKey<String> DB_SYSTEM = AttributeKey.stringKey("db.system");
+  private static final AttributeKey<List<String>> AWS_DYNAMODB_TABLE_NAMES =
+      AttributeKey.stringArrayKey("aws.dynamodb.table_names");
+
+  private static final String DYNAMODB = "dynamodb";
+
+  @Override
+  public void onStart(AttributesBuilder attributes, Context parentContext, Request<?> request) {
+    AttributesExtractorUtil.internalSet(attributes, DB_SYSTEM, DYNAMODB);
+    String tableName = RequestAccess.getTableName(request.getOriginalRequest());
+    AttributesExtractorUtil.internalSet(
+        attributes, AWS_DYNAMODB_TABLE_NAMES, Collections.singletonList(tableName));
```

Comment:
this is just an observation more than a request for a change - I understand that this attribute is a `List<String>` based on the [spec](https://opentelemetry.io/docs/specs/semconv/database/dynamodb/#dynamodbbatchgetitem), but it doesn't look like this code supports multiple table names, so it feels a little strange for it be defined that way

Reply:
Yeah we can likely make an additional enhancement that creates a Request Accessor for the methods that are tied to table names for the batch functions, and then add that Request Accessor to the logic for setting the value. Right now getTable is tied to getTableName which will always return one.